add tqdm progress bar for faithfullness metrics

dkrako opened the pull request:
For my own projects I am applying the faithfulness metrics on the complete data set.
As this can take quite long, I added an optional tqdm progress bar for this.
It is turned off by default (`disable_progress_bar=False`) to avoid unwanted changes to existing workflows.

I haven't added any tests for this. I think it should be sufficient to just run all faithfulness metric tests with `disable_progress_bar=True` and `disable_progress_bar=False` for full test coverage.
I'll add the additional fixture parameters to `test_faithfulness_metrics.py` if you would prefer to have this feature tested. I'm just not quite sure how to do this correctly.

Please let me know if you don't want any progress bars included or think that running the metrics on the complete data set is a bad idea and batched execution should be preferred.

annahedstroem commented on quantus/metrics/faithfulness_metrics.py:
```diff
@@ -178,7 +180,14 @@ def __call__(
+        if self.disable_progress_bar:
+            iterator = zip(x_batch_s, y_batch, a_batch)
+        else:
+            from tqdm import tqdm
```
can you please put the imports at top of the scripts? the import should be placed before relative imports like import ..helpers.norm_func

dkrako replied:
ah sorry, yes of course. (this was just a leftover from when I thought I would implement it as an optional package dependency. but as the use of tqdm is so widespread and the package is so tiny, I decided to just do this the disabled-by-default-way)

annahedstroem reviewed (commented):
Hi @dkrako 

I really like the addition of the progress bar into the metric implementation and also, it was right to set the default to make sure the progress bar is disabled.

Would you mind also adding a progress bar similarly in the other metrics in Quantus? I think it can be nice to have it implemented across the board

Thanks.

annahedstroem reviewed (commented):
thanks for adding pytests :-)

dkrako commented:
I renamed the parameter `disable_progress_bar` to the more intuitive parameter `display_progressbar`.

I also added the progress bar to the outer loop of the `ModelParameterRandomisation`. I needed an additional call of`model.get_random_layer_generator`, to be able to know beforehand how many layer-iterations there will be. I think this shouldn't be a problem. I didn't want to create an additional method in the model adapters for tensorflow and pytorch.

dkrako commented:
@annahedstroem I think this feature is ready for final code review and merging. Overall test coverage increased by 0.12% :)

annahedstroem commented:
Thanks so much @dkrako for this contribution! (and for bumping up the test coverage - awesome).

- `disable_progressbar`: change of name is better, I agree
- solution for `model.get_random_layer_generator` sounds wise

Changes accepted and merged 👍
